`expect` triggers incompatibility with mocha declarations

<!-- Love Jest? Please consider supporting our collective: 👉  https://opencollective.com/jest/donate -->

## 🐛 Bug Report

Trying to use the `expect` assertion library on a mocha project causes TypeScript errors, even though there should not be any incompatibility.

Both `@types/mocha` and `@jest/types` try to declare incompatible global TS declarations for `it`, `describe`, etc, which causes TS compiler errors.  Obviously users should not try to load both of these at once.  *However*, the `expect` type declarations refer to `@jest/types`.  This means it's impossible to use the `expect` assertion library in a project that uses mocha as its test runner.

I think the solution is straightforward:

1. Refactor `@jest/types` so it's possible to import `@jest/types/no-global` which will not augment global interfaces.
2. Refactor `expect` to import types from `@jest/types/no-global` instead of `@jest/types`

This can be fully backwards compatible.

## To Reproduce

Steps to reproduce the behavior:

```
yarn add @types/node expect @types/mocha typescript
echo "import expect from 'expect';" > index.ts
./node_modules/.bin/tsc --init
```
Set `"types": ["mocha"]` in the tsconfig file to avoid eagerly loading unrelated `@types`.
```
./node_modules/.bin/tsc --build
```

You'll see the following TypeScript error:

```
node_modules/@types/node/globals.d.ts:1032:15 - error TS2430: Interface 'Global' incorrectly extends interface 'MochaGlobals'.
  Types of property 'describe' are incompatible.
    Type 'Describe' is not assignable to type 'SuiteFunction'.
      Types of property 'only' are incompatible.
        Type 'DescribeBase' is not assignable to type 'ExclusiveSuiteFunction'.
          Type 'void' is not assignable to type 'Suite'.

1032     interface Global {
                   ~~~~~~


Found 1 error.
```

## Expected behavior

Importing `expect` in a TS project does not augment global interfaces with Jest-specific type declarations.

## Link to repl or repo (highly encouraged)

https://repl.it/repls/SpanishHugeVirus

## Run `npx envinfo --preset jest`

Paste the results here:

```bash
 System:
    OS: Linux 4.9 Pengwin undefined
    CPU: (4) x64 Intel(R) Core(TM) i7-6600U CPU @ 2.60GHz
  Binaries:
    Node: 10.15.2 - /usr/bin/node
    Yarn: 1.16.0 - /usr/bin/yarn
    npm: 6.9.0 - ~/n/bin/npm
```

--- a/packages/jest-types/src/Global.ts
+++ b/packages/jest-types/src/Global.ts
@@ -67,7 +67,7 @@ export interface Describe extends DescribeBase {
 }
 
 // TODO: Maybe add `| Window` in the future?
-export interface Global extends NodeJS.Global {
+export interface GlobalAdditions {
   it: ItConcurrent;
   test: ItConcurrent;
   fit: ItBase & {concurrent?: ItConcurrentBase};
@@ -82,5 +82,9 @@ export interface Global extends NodeJS.Global {
   pending: () => void;
   spyOn: () => void;
   spyOnProperty: () => void;
+}
+export interface Global
+  extends GlobalAdditions,
+    Omit<NodeJS.Global, keyof GlobalAdditions> {
   [extras: string]: any;
 }

